### PR TITLE
Python wheel server

### PR DESCRIPTION
**Pull request by JoFrost**
This PR adds a linux version of BlindAI and a Python wheel to execute this server. A simulation mode support was added as well to the client.

The Python wheel was mainly made to facilitate the test of BlindAI, allowing to run a local copy of the server in a Google Colab/Jupyter Notebook environment. Considering that the python wheel will not run BlindAI in an SGX environment, the simulation mode is necessary to test this version of the server.

Running the server is very simple, once the wheel is installed: 
```
import blindai_preview_server

srv = blindai_preview_server.start() # start the server
blindai_preview_server.stop(srv) # stop the server
```
This PR includes the necessary CI steps to build automatically the wheel and the server when a new tag is available.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -68,11 +68,11 @@ check *args:
 
 # Build for a Linux target (no SGX)
 build-no-sgx *args:
-  cargo build {{args}}
+  cargo build --target x86_64-unknown-linux-gnu {{args}}
 
 # Run on a Linux target (no SGX)
 run-no-sgx *args:
-  cargo run {{args}}
+  cargo run --target x86_64-unknown-linux-gnu {{args}}
```
Comment (clauverjat) [inline justfile:75]: Don't need that. The local target is the default.

File: client/blindai_preview/_dcap_attestation.py [hunk 2]
```diff
@@ -116,6 +116,9 @@ def validate_attestation(
         __package__, "Intel_SGX_Provisioning_Certification_RootCA.pem"
     )
 
+    if len(collateral) < 1:
```
Comment (clauverjat) [inline client/blindai_preview/_dcap_attestation.py:119]: It is a bit strange.
We might want to check that the collateral structure from the server match the one that is expected. But as it is it is only a partial check. What if the server removes a field ? 
In short, we might want to check, but if we check we need to do it fully.

File: client/pyproject.toml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [tool.poetry]
 name = "blindai-preview"
-version = "0.0.5"
+version = "0.0.6"
 description = ""
-authors = ["Corentin Lauverjat <corentin.lauverjat@mithrilsecurity.io>"]
+authors = ["Mithril Security <contact@mithrilsecurity.io>"]
```
Comment (clauverjat) [inline client/pyproject.toml:5]: Good catch

File: src/main.rs [hunk 4]
```diff
@@ -118,14 +118,28 @@ fn main() -> Result<()> {
     let mut report_data = [0u8; 64];
     report_data[0..32].copy_from_slice(report_binding.as_ref());
```
Comment (clauverjat) [inline src/main.rs:120]: Hard to follow the logic with all the compile flags

File: client/blindai_preview/_dcap_attestation.py (hunk 2 above)
Comment (JoFrost) [reply client/blindai_preview/_dcap_attestation.py:119]: This is a hack, but the client will simply fail if the collateral is empty, and will not return a proper exception. I had this issue when testing the simulation mode, as the simulation mode will send empty collateral...

File: justfile (hunk 1 above)
Comment (JoFrost) [reply justfile:75]: It is actually needed because the rust-toolchain.toml does specify the fortanix toolchain, I am simply forcing to use the original linux toolchain

File: src/main.rs (hunk 4 above)
Comment (JoFrost) [reply src/main.rs:120]: I am simply disabling the attestation/collateral creation if we are not in an SGX environment. I had no choice than specifying the compile flags like this.

File: client/blindai_preview/_dcap_attestation.py (hunk 2 above)
Comment (JoFrost) [reply client/blindai_preview/_dcap_attestation.py:119]: I wanted at first to check if the exact amount of data are available, but to keep it simple, I am now simply checking this. Anyway, if one field is missing, the client will just crash, atm.

File: src/main.rs (hunk 4 above)
Comment (JoFrost) [reply src/main.rs:120]: But I guess it can be refactored

File: justfile (hunk 1 above)
Comment (clauverjat) [reply justfile:75]: I kind of get why this can be confusing, but default toolchain != default target. It works perfectly on my end without the "--target x86_64-unknown-linux-gnu". I'd be curious to know in what configuration you were for sgx to be the default target.

File: client/blindai_preview/_dcap_attestation.py (hunk 2 above)
Comment (clauverjat) [reply client/blindai_preview/_dcap_attestation.py:119]: I see. We need to change that, but I think we can do it in another PR/commit.

**Comment by clauverjat** [comment]
Closing since #16 is merged.
